monitor/monitor.go

```diff
@@ -0,0 +1,260 @@
+package monitor
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/9corp/9volt/alerter"
+	"github.com/9corp/9volt/config"
+	"github.com/9corp/9volt/dal"
+	"github.com/9corp/9volt/util"
+)
+
+const (
+	START int = iota
+	STOP
+
+	GOROUTINE_ID_LENGTH = 8
+)
+
+type IMonitor interface {
+	Run() error
+	Stop()
+	Identify() string
+}
+
+type Monitor struct {
+	Config             *config.Config
+	Identifier         string
+	runningMonitorLock *sync.Mutex
+	runningMonitors    map[string]IMonitor
+	MessageChannel     chan *alerter.Message
+	SupportedMonitors  map[string]func(*RootMonitorConfig) IMonitor // monitor name : NewXMonitor
+}
+
+type RootMonitorConfig struct {
+	GID            string // goroutine id
+	Name           string
+	Config         *MonitorConfig
+	MessageChannel chan *alerter.Message
+	StopChannel    chan bool
+	Ticker         *time.Ticker
+}
+
+// TODO: This should probably be split up between each individual check type
+type MonitorConfig struct {
```

Comment:
Not a fan of this single gigantic struct but I dunno if there's any gain from splitting it up further; seems like it would only introduce complexity.